src/TreeSupport.cpp

```diff
@@ -650,32 +610,119 @@ void TreeSupport::insertDroppedNode(std::unordered_set<Node*>& nodes_layer, Node
     conflicting_node->support_roof_layers_below = std::max(conflicting_node->support_roof_layers_below, p_node->support_roof_layers_below);
 }
 
-void TreeSupport::propagateCollisionAreas(const SliceDataStorage& storage, const std::vector<std::vector<Polygons>>& model_collision, std::vector<std::vector<Polygons>>& model_avoidance)
+ModelVolumes::ModelVolumes(const SliceDataStorage& storage, Polygons machine_border, coord_t xy_distance,
```

Comment:
`machine_border` is constructed in the `TreeSupport` constructor and doesnt live beyond that so using a reference or pointer wont work.

Passing by value and moving allows some compiler optimisation since it knows it can modify the passed in value. What i forgot to do was actually call std::move, i'll add it in the appropriate places. Theres a decent chance the compiler will do a move anyway but better safe than sorry

Reply:
Actually I think you should move the whole object into your new class.
It's only used for the stuff which that class is responsible for.

Reply:
Thats a good point. I initially tried to modify the TreeSupport class as little as possible but it doesnt make sense for it to be outside ModelVolumes

I'll move it in.

Reply:
I think that in this instance the machine border can be thought of as a special model which is always there.

I'm not sure where you are putting it now, but it would be a bit too much to make a separate file for it - if that's what you were planning.

I would be okay with the code being in SliceDataStorage somewhere.

Reply:
Ok, i've moved the machine border code to SliceDataStorage and the machine border collision area to a private static function in ModelVolumes